abel/tools.py

```diff
@@ -195,50 +301,163 @@ def center_image_asym(data, center_column, n_vert, n_horz, verbose=False):
+    right = right[rmin:rmax]
+    top   = top[::-1][rmin:rmax]    # flip
+    bot   = bot[rmin:rmax]
+
+    # compare and determine shift to best overlap slice profiles
+    # fix me! - should be consistent and use curve_fit()?
+    shift0 = [0,]
+    if center_horizontal:
+        horiz = minimize(align, shift0, args=(left, right))
+    else: 
+        horiz['x'] = 0.0
+   
+    if center_vertical:
+        vert  = minimize(align, shift0, args=(top, bot))
+    else:
+        vert['x'] = 0.0
+
+    col_shift = horiz['x']
+    row_shift = vert['x']
+
+    if pixel_center: 
+        col_shift += 0.5
+        row_shift -= 0.5
+
+    IM_centered = shift(IM,(row_shift,col_shift)) # center image
+
+    if rows%2==0 and pixel_center:    # make odd size image
+        IM_centered = IM_centered[:-1, 1:]  # drop left most column, bottom row
+
+    return IM_centered, (row_shift,col_shift)
+
+
 # The next two functions are adapted from
 # http://stackoverflow.com/questions/3798333/image-information-along-a-polar-coordinate-system
 # It is possible that there is a faster way to convert to polar coordinates.
 
-def reproject_image_into_polar(data, origin=None):
+def reproject_image_into_polar(data, origin=None, Jacobian=False,
+                               rmax='corner', dr=1, dt=None):
     """Reprojects a 2D numpy array ("data") into a polar coordinate system.
     "origin" is a tuple of (x0, y0) and defaults to the center of the image.
+    
+    Parameters
+    ----------
+     - data:   rowsxcolums numpy array
+     - origin: tuple, the coordinate of the image center
+     - Jacobian: boolean, include r intensity scaling in the coordinate transform
+     - rmax:   radius maximum, beyond which no projection is done
+               'corner' => sqrt((rows//2)**2+(cols//2)**2)
+                  - used to convert the whole cartesian image to polar
+               otherwise the minimum of the 1/2 width or 1/2 height of the image
+               = min((rows//2, cols//2)
+                  - used to convert only the circular region that fits within
+                    the image
+     - dr: radial coordinate spacing for the grid interpolation
+             tests show that there is not much point in going below 0.5
+     - dt: angular coordinate spacing (in degrees)
+
+     Returns
+     -------
+      - output:     rows x cols  or row(col)xrow(col) numpy array, polar image
+      - r_grid:     meshgrid of radial coordinates
+      - theta_grid: meshgrid of theta coordinates
     """
     ny, nx = data.shape[:2]
     if origin is None:
-        origin = (nx//2, ny//2)
+        origin = (nx//2+nx%2, ny//2+ny%2)   # % handles odd size image
 
     # Determine that the min and max r and theta coords will be...
     x, y = index_coords(data, origin=origin)
-    r, theta = cart2polar(x, y)
+    r, theta = cart2polar(x, y)   # convert (x,y) -> (r,θ)
 
-    nr = r.max()
-    nt = ny//2
+    if rmax == 'corner':
+       rmax = r.max()
+       nr = rmax 
```

Comment:
``nr`` needs to take into account ``nr``. At a minimum, this should be 
``nr = rmax/dr``,
but perhaps more complicated as it is below?